src/terminator/mod.rs

```diff
@@ -559,6 +559,36 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
         let usize = self.tcx.types.usize;
 
         match &link_name[..] {
+            "malloc" => {
+                let size = self.value_to_primval(args[0], usize)?.to_u64()?;
+                let align = self.memory.pointer_size();
+                let ptr = self.memory.allocate(size, align)?;
```

Comment:
If size is `0`, this will do the wrong thing. The spec says `malloc(0)` must either return NULL or return something that can be passed to `free`.

(Btw, it seems like the `__rust` allocation functions also don't treat size `0` specially -- technically, I believe they are all UB, so whatever miri does is fine, but it'd be nicer to detect that UB.)